main.go

```diff
@@ -165,13 +163,16 @@ func (s *prometheusServer) Start(ctx context.Context) error {
 
 	server.TLSConfig = s.getPrometheusTLSConfig(ctx, certWatcher)
 
-	if err := server.ListenAndServeTLS(s.certPath, s.keyPath); err != nil && !errors.Is(err, http.ErrServerClosed) {
+	if err = server.ListenAndServeTLS(s.certPath, s.keyPath); err != nil && !errors.Is(err, http.ErrServerClosed) {
 		setupLog.Error(err, "Failed to start Prometheus metrics endpoint server")
-		return err
+		cancel()
 	}
 
 	<-idleConnsClosed
-	return nil
+	if errors.Is(err, http.ErrServerClosed) {
```

Comment:
It is usual that the last return in a function is the successful path `return nil`. Can you invert this `if` so that it returns the error?